lib/reek/configuration/app_configuration.rb

```diff
@@ -11,135 +15,89 @@ module Configuration
+      private_attr_accessor :directory_directives, :default_directive, :excluded_paths
 
-      # Given this configuration file:
+      # Instantiate a configuration via given path.
       #
-      # ---
-      # IrresponsibleModule:
-      #   enabled: false
-      # "app/helpers":
-      #   UtilityFunction:
-      #     enabled: false
-      # exclude_paths:
-      #   - "app/controllers"
+      # @param path [Pathname] the path to the config file
       #
-      # this would result in the following configuration:
-      #
-      # exclude_paths = [ Pathname('spec/samples/two_smelly_files') ]
-      # default_directive = { Reek::Smells::IrresponsibleModule => { "enabled" => false } }
-      # directory_directives = { Pathname("spec/samples/three_clean_files/") =>
-      #                          { Reek::Smells::UtilityFunction => { "enabled" => false } } }
+      # @return [AppConfiguration]
+      def self.from_path(path: nil)
+        instance = allocate
+        instance.instance_eval do
+          self.directory_directives = {}.extend(DirectoryDirectives)
+          self.default_directive    = {}.extend(DefaultDirective)
+          self.excluded_paths       = [].extend(ExcludedPaths)
+
+          find_and_load(path: path)
+        end
+        instance
+      end
+
+      # Instantiate a configuration by passing everything in.
       #
+      # @param directory_directives [Hash] for instance:
+      #   { Pathname("spec/samples/three_clean_files/") =>
+      #     { Reek::Smells::UtilityFunction => { "enabled" => false } } }
+      # @param default_directive [Hash] for instance:
+      #   { Reek::Smells::IrresponsibleModule => { "enabled" => false } }
+      # @param excluded_paths [Array] for instance:
+      #   [ Pathname('spec/samples/two_smelly_files') ]
       #
-      # @param options [OpenStruct]
-      #   e.g.: #<OpenStruct report_format =: text,
-      #                      location_format =: numbers,
-      #                      colored = true,
-      #                      smells_to_detect = [],
-      #                      config_file = #<Pathname:config/defaults.reek>,
-      #                      argv = [ "lib/reek/spec" ]>
-      def initialize(options = nil)
-        @directory_directives = {}
-        @default_directive    = {}
-        @exclude_paths        = []
+      # @return [AppConfiguration]
+      def self.from_map(directory_directives: {},
+                        default_directive: {},
+                        excluded_paths: [])
+        instance = allocate
+        instance.instance_eval do
+          self.directory_directives = directory_directives.extend(DirectoryDirectives)
+          self.default_directive    = default_directive.extend(DefaultDirective)
+          self.excluded_paths       = excluded_paths.extend(ExcludedPaths)
+        end
+        instance
+      end
+
+      def self.default
+        from_path path: nil
+      end
```

Comment:
`.from_path`’s parameter defaults to `nil`, so this could just call a bare `.from_path`. Not sure if more readable, though…